pkg/ebpf/dns_parser.go

```diff
@@ -41,7 +43,7 @@ func newDNSParser() *dnsParser {
 func (p *dnsParser) ParseInto(data []byte, t *translation) error {
 	err := p.decoder.DecodeLayers(data, &p.layers)
 	if err != nil || p.decoder.Truncated {
-		return nil
```

Comment:
We mentioned it in standup - but just adding here so it's tracked. 

A separate counter should be reported/incremented here for # of truncated responses.

Reply:
Both in the case of TCP fragmentation (`dig +dnssec @a.gtld-servers.net com. any`) and IP fragmentation (the scenario created by adding many A records), this `Truncated` field was not set 😞. So we cannot really use this value to differentiate between a garbled packet and a fragmented one.

We can use the exact error messages  to differentiate TCP segmentation related errors (e.g. `errDecodeRecordLength`, `errDNSNameOffsetNegative`, etc. ) 
 based on https://github.com/google/gopacket/blob/master/layers/dns.go#L1063 but I don't know the likelihood of encountering these errors in other invalid but nonfragmented scenarios. 

Also, in the case of segmentation, there is no way to know if it contained valid A records or not since we will run into Parsing error in an earlier step. For example `dig +dnssec @a.gtld-servers.net com. any` results in A records being sent as UDP packets and the segmented TCP packets do not contain anything relevant (figured out that when in some cases the TCP packets were not getting segmented). So even if we can't parse them, it does not affect our DNS snooping capability in anyway.